core/src/main/java/com/netease/arctic/utils/map/StructLikeWrapperSizeEstimator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.utils.map;
+
+import com.netease.arctic.iceberg.optimize.StructLikeWrapper;
+import com.netease.arctic.utils.ObjectSizeCalculator;
+
+/**
+ * Size Estimator for StructLikeWrapper record payload.
+ */
+public class StructLikeWrapperSizeEstimator implements SizeEstimator<StructLikeWrapper> {
+  @Override
+  public long sizeEstimate(StructLikeWrapper structLikeWrapper) {
+    return ObjectSizeCalculator.getObjectSize(structLikeWrapper.get());
```

Comment:
If `structLikeWrapper == null`, it will incur NPE.

Reply:
OK